<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>16 Bibliography | Comparative Methods Workshops</title>
  <meta name="description" content="These are the workshops for the Comparative Research Group as part of LIFE363" />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="16 Bibliography | Comparative Methods Workshops" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="These are the workshops for the Comparative Research Group as part of LIFE363" />
  <meta name="github-repo" content="ChrisMitchell42/bookdownCRG" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="16 Bibliography | Comparative Methods Workshops" />
  
  <meta name="twitter:description" content="These are the workshops for the Comparative Research Group as part of LIFE363" />
  

<meta name="author" content="Chris Mitchell" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="tips-and-tricks.html"/>

<script src="libs/header-attrs-2.3/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/iframe-resizer/3.5.16/iframeResizer.min.js" type="text/javascript"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Comparative Research Group</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Welcome</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#for-students"><i class="fa fa-check"></i><b>1.1</b> For students</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#what-is-the-comparative-method"><i class="fa fa-check"></i><b>2.1</b> What is the comparative method?</a></li>
<li class="chapter" data-level="2.2" data-path="intro.html"><a href="intro.html#tree-thinking"><i class="fa fa-check"></i><b>2.2</b> Tree thinking</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="recap.html"><a href="recap.html"><i class="fa fa-check"></i><b>3</b> R recap</a>
<ul>
<li class="chapter" data-level="3.1" data-path="recap.html"><a href="recap.html#basics"><i class="fa fa-check"></i><b>3.1</b> Basics</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="recap.html"><a href="recap.html#plotting"><i class="fa fa-check"></i><b>3.1.1</b> Plotting</a></li>
<li class="chapter" data-level="3.1.2" data-path="recap.html"><a href="recap.html#the-working-directory"><i class="fa fa-check"></i><b>3.1.2</b> The working directory</a></li>
<li class="chapter" data-level="3.1.3" data-path="recap.html"><a href="recap.html#loading-data"><i class="fa fa-check"></i><b>3.1.3</b> Loading data</a></li>
<li class="chapter" data-level="3.1.4" data-path="recap.html"><a href="recap.html#subsetting"><i class="fa fa-check"></i><b>3.1.4</b> Subsetting</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="recap.html"><a href="recap.html#errors"><i class="fa fa-check"></i><b>3.2</b> Errors</a></li>
<li class="chapter" data-level="3.3" data-path="recap.html"><a href="recap.html#google"><i class="fa fa-check"></i><b>3.3</b> Google</a></li>
<li class="chapter" data-level="3.4" data-path="recap.html"><a href="recap.html#stealing"><i class="fa fa-check"></i><b>3.4</b> Stealing</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="phylogenetics.html"><a href="phylogenetics.html"><i class="fa fa-check"></i><b>4</b> Phylogenetic trees and where to find them</a>
<ul>
<li class="chapter" data-level="4.1" data-path="phylogenetics.html"><a href="phylogenetics.html#phylogeny"><i class="fa fa-check"></i><b>4.1</b> Phylogeny</a></li>
<li class="chapter" data-level="4.2" data-path="phylogenetics.html"><a href="phylogenetics.html#building-trees"><i class="fa fa-check"></i><b>4.2</b> Building trees</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="phylogenetics.html"><a href="phylogenetics.html#morphology"><i class="fa fa-check"></i><b>4.2.1</b> Morphology</a></li>
<li class="chapter" data-level="4.2.2" data-path="phylogenetics.html"><a href="phylogenetics.html#molecular-data"><i class="fa fa-check"></i><b>4.2.2</b> Molecular data</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="phylogenetics.html"><a href="phylogenetics.html#tree-structure"><i class="fa fa-check"></i><b>4.3</b> Tree structure</a></li>
<li class="chapter" data-level="4.4" data-path="phylogenetics.html"><a href="phylogenetics.html#finding-trees"><i class="fa fa-check"></i><b>4.4</b> Finding trees</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="phylogenetics.html"><a href="phylogenetics.html#vertlife"><i class="fa fa-check"></i><b>4.4.1</b> Vertlife</a></li>
<li class="chapter" data-level="4.4.2" data-path="phylogenetics.html"><a href="phylogenetics.html#ktrees"><i class="fa fa-check"></i><b>4.4.2</b> 10ktrees</a></li>
<li class="chapter" data-level="4.4.3" data-path="phylogenetics.html"><a href="phylogenetics.html#treebase"><i class="fa fa-check"></i><b>4.4.3</b> TreeBASE</a></li>
<li class="chapter" data-level="4.4.4" data-path="phylogenetics.html"><a href="phylogenetics.html#primary-literature"><i class="fa fa-check"></i><b>4.4.4</b> Primary literature</a></li>
<li class="chapter" data-level="4.4.5" data-path="phylogenetics.html"><a href="phylogenetics.html#tree.drawer"><i class="fa fa-check"></i><b>4.4.5</b> tree.drawer</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="phylogenetics.html"><a href="phylogenetics.html#further-info"><i class="fa fa-check"></i><b>4.5</b> Further info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="w1plotting.html"><a href="w1plotting.html"><i class="fa fa-check"></i><b>5</b> Plotting trees in R</a>
<ul>
<li class="chapter" data-level="5.1" data-path="w1plotting.html"><a href="w1plotting.html#phylogenies-in-r"><i class="fa fa-check"></i><b>5.1</b> Phylogenies in R</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="w1plotting.html"><a href="w1plotting.html#packages-used"><i class="fa fa-check"></i><b>5.1.1</b> Packages used</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="w1plotting.html"><a href="w1plotting.html#importing-your-tree"><i class="fa fa-check"></i><b>5.2</b> Importing your tree</a></li>
<li class="chapter" data-level="5.3" data-path="w1plotting.html"><a href="w1plotting.html#ggtree"><i class="fa fa-check"></i><b>5.3</b> ggtree</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="w1plotting.html"><a href="w1plotting.html#geoms"><i class="fa fa-check"></i><b>5.3.1</b> Geoms</a></li>
<li class="chapter" data-level="5.3.2" data-path="w1plotting.html"><a href="w1plotting.html#labelling-clades"><i class="fa fa-check"></i><b>5.3.2</b> Labelling clades</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="w1plotting.html"><a href="w1plotting.html#adding-images-to-trees"><i class="fa fa-check"></i><b>5.4</b> Adding images to trees</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="w1plotting.html"><a href="w1plotting.html#phylopic"><i class="fa fa-check"></i><b>5.4.1</b> Phylopic</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="w1plotting.html"><a href="w1plotting.html#further-info-1"><i class="fa fa-check"></i><b>5.5</b> Further info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>6</b> Data</a>
<ul>
<li class="chapter" data-level="6.1" data-path="data.html"><a href="data.html#comparative-data"><i class="fa fa-check"></i><b>6.1</b> Comparative data</a></li>
<li class="chapter" data-level="6.2" data-path="data.html"><a href="data.html#databases"><i class="fa fa-check"></i><b>6.2</b> Databases</a></li>
<li class="chapter" data-level="6.3" data-path="data.html"><a href="data.html#data-riches"><i class="fa fa-check"></i><b>6.3</b> Data Riches</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="anova.html"><a href="anova.html"><i class="fa fa-check"></i><b>7</b> ANOVA</a>
<ul>
<li class="chapter" data-level="7.1" data-path="anova.html"><a href="anova.html#analysis-of-variance"><i class="fa fa-check"></i><b>7.1</b> Analysis of variance</a></li>
<li class="chapter" data-level="7.2" data-path="anova.html"><a href="anova.html#phylogenetic-correction"><i class="fa fa-check"></i><b>7.2</b> Phylogenetic correction</a></li>
<li class="chapter" data-level="7.3" data-path="anova.html"><a href="anova.html#example-data-analysis"><i class="fa fa-check"></i><b>7.3</b> Example data &amp; analysis</a></li>
<li class="chapter" data-level="7.4" data-path="anova.html"><a href="anova.html#further-info-2"><i class="fa fa-check"></i><b>7.4</b> Further info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="asr1.html"><a href="asr1.html"><i class="fa fa-check"></i><b>8</b> Ancestral State Reconstruction I</a>
<ul>
<li class="chapter" data-level="8.1" data-path="asr1.html"><a href="asr1.html#data-1"><i class="fa fa-check"></i><b>8.1</b> Data</a></li>
<li class="chapter" data-level="8.2" data-path="asr1.html"><a href="asr1.html#trees"><i class="fa fa-check"></i><b>8.2</b> Trees</a></li>
<li class="chapter" data-level="8.3" data-path="asr1.html"><a href="asr1.html#parsimony"><i class="fa fa-check"></i><b>8.3</b> Parsimony</a></li>
<li class="chapter" data-level="8.4" data-path="asr1.html"><a href="asr1.html#maximum-likelihood"><i class="fa fa-check"></i><b>8.4</b> Maximum Likelihood</a></li>
<li class="chapter" data-level="8.5" data-path="asr1.html"><a href="asr1.html#stochastic-character-mapping"><i class="fa fa-check"></i><b>8.5</b> Stochastic Character Mapping</a>
<ul>
<li class="chapter" data-level="8.5.1" data-path="asr1.html"><a href="asr1.html#an-analogy"><i class="fa fa-check"></i><b>8.5.1</b> An Analogy</a></li>
<li class="chapter" data-level="8.5.2" data-path="asr1.html"><a href="asr1.html#state-characters"><i class="fa fa-check"></i><b>8.5.2</b> 2-State Characters</a></li>
<li class="chapter" data-level="8.5.3" data-path="asr1.html"><a href="asr1.html#state-characters-1"><i class="fa fa-check"></i><b>8.5.3</b> 3-State Characters</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="asr1.html"><a href="asr1.html#further-info-3"><i class="fa fa-check"></i><b>8.6</b> Further info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="asr2.html"><a href="asr2.html"><i class="fa fa-check"></i><b>9</b> Ancestral State Reconstruction II</a>
<ul>
<li class="chapter" data-level="9.1" data-path="asr2.html"><a href="asr2.html#data-3"><i class="fa fa-check"></i><b>9.1</b> Data</a></li>
<li class="chapter" data-level="9.2" data-path="asr2.html"><a href="asr2.html#tree"><i class="fa fa-check"></i><b>9.2</b> Tree</a></li>
<li class="chapter" data-level="9.3" data-path="asr2.html"><a href="asr2.html#ancestral-state-reconstructions"><i class="fa fa-check"></i><b>9.3</b> Ancestral State Reconstructions</a></li>
<li class="chapter" data-level="9.4" data-path="asr2.html"><a href="asr2.html#bayestraits"><i class="fa fa-check"></i><b>9.4</b> BayesTraits</a></li>
<li class="chapter" data-level="9.5" data-path="asr2.html"><a href="asr2.html#modelling-evolution"><i class="fa fa-check"></i><b>9.5</b> Modelling Evolution</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="asr2.html"><a href="asr2.html#brownian-motion"><i class="fa fa-check"></i><b>9.5.1</b> Brownian Motion</a></li>
<li class="chapter" data-level="9.5.2" data-path="asr2.html"><a href="asr2.html#directional-evolution"><i class="fa fa-check"></i><b>9.5.2</b> Directional Evolution</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="asr2.html"><a href="asr2.html#changes-in-the-rate-of-evolution-of-a-trait"><i class="fa fa-check"></i><b>9.6</b> Changes in the rate of evolution of a trait</a></li>
<li class="chapter" data-level="9.7" data-path="asr2.html"><a href="asr2.html#uncertainty"><i class="fa fa-check"></i><b>9.7</b> Uncertainty</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="asr2.html"><a href="asr2.html#fossils"><i class="fa fa-check"></i><b>9.7.1</b> Fossils</a></li>
<li class="chapter" data-level="9.7.2" data-path="asr2.html"><a href="asr2.html#revisiting-mysticete-body-mass"><i class="fa fa-check"></i><b>9.7.2</b> Revisiting Mysticete Body Mass</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="asr2.html"><a href="asr2.html#further-info-4"><i class="fa fa-check"></i><b>9.8</b> Further info</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="diversification.html"><a href="diversification.html"><i class="fa fa-check"></i><b>10</b> Diversification</a>
<ul>
<li class="chapter" data-level="10.1" data-path="diversification.html"><a href="diversification.html#introduction"><i class="fa fa-check"></i><b>10.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="diversification.html"><a href="diversification.html#pure-birth-model"><i class="fa fa-check"></i><b>10.1.1</b> Pure Birth Model</a></li>
<li class="chapter" data-level="10.1.2" data-path="diversification.html"><a href="diversification.html#the-birth-death-model"><i class="fa fa-check"></i><b>10.1.2</b> The Birth-Death Model</a></li>
<li class="chapter" data-level="10.1.3" data-path="diversification.html"><a href="diversification.html#the-signature-of-extinction"><i class="fa fa-check"></i><b>10.1.3</b> The signature of extinction</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="diversification.html"><a href="diversification.html#estimating-speciation-and-extinction-rate"><i class="fa fa-check"></i><b>10.2</b> Estimating speciation and extinction rate</a></li>
<li class="chapter" data-level="10.3" data-path="diversification.html"><a href="diversification.html#medusa"><i class="fa fa-check"></i><b>10.3</b> MEDUSA</a></li>
<li class="chapter" data-level="10.4" data-path="diversification.html"><a href="diversification.html#rpanda"><i class="fa fa-check"></i><b>10.4</b> RPANDA</a></li>
<li class="chapter" data-level="10.5" data-path="diversification.html"><a href="diversification.html#bamm-bammtools"><i class="fa fa-check"></i><b>10.5</b> BAMM &amp; BAMMtools</a></li>
<li class="chapter" data-level="10.6" data-path="diversification.html"><a href="diversification.html#further-info-5"><i class="fa fa-check"></i><b>10.6</b> Further info</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="diversification2.html"><a href="diversification2.html"><i class="fa fa-check"></i><b>11</b> Trait Dependent Diversification</a>
<ul>
<li class="chapter" data-level="11.1" data-path="diversification2.html"><a href="diversification2.html#binary-traits-bisse"><i class="fa fa-check"></i><b>11.1</b> Binary traits: BiSSE</a></li>
<li class="chapter" data-level="11.2" data-path="diversification2.html"><a href="diversification2.html#multi-state-traits-musse"><i class="fa fa-check"></i><b>11.2</b> Multi-state traits: MuSSE</a></li>
<li class="chapter" data-level="11.3" data-path="diversification2.html"><a href="diversification2.html#quantitative-traits-quasse"><i class="fa fa-check"></i><b>11.3</b> Quantitative traits: QuaSSE</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="diversification2.html"><a href="diversification2.html#quasse-primate-example"><i class="fa fa-check"></i><b>11.3.1</b> QuaSSE primate example</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="w2PGLS.html"><a href="w2PGLS.html"><i class="fa fa-check"></i><b>12</b> Phylogenetic Regression</a>
<ul>
<li class="chapter" data-level="12.1" data-path="w2PGLS.html"><a href="w2PGLS.html#data-4"><i class="fa fa-check"></i><b>12.1</b> Data</a></li>
<li class="chapter" data-level="12.2" data-path="w2PGLS.html"><a href="w2PGLS.html#linear-regression"><i class="fa fa-check"></i><b>12.2</b> Linear Regression</a></li>
<li class="chapter" data-level="12.3" data-path="w2PGLS.html"><a href="w2PGLS.html#phylogenetic-signal"><i class="fa fa-check"></i><b>12.3</b> Phylogenetic Signal</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="w2PGLS.html"><a href="w2PGLS.html#phylogenetic-signal-1"><i class="fa fa-check"></i><b>12.3.1</b> Phylogenetic signal</a></li>
<li class="chapter" data-level="12.3.2" data-path="w2PGLS.html"><a href="w2PGLS.html#caper"><i class="fa fa-check"></i><b>12.3.2</b> caper</a></li>
<li class="chapter" data-level="12.3.3" data-path="w2PGLS.html"><a href="w2PGLS.html#estimating-phylogenetic-signal"><i class="fa fa-check"></i><b>12.3.3</b> Estimating Phylogenetic Signal</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="w2PGLS.html"><a href="w2PGLS.html#phylogenetic-regression"><i class="fa fa-check"></i><b>12.4</b> Phylogenetic Regression</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="w2PGLS.html"><a href="w2PGLS.html#model-checking"><i class="fa fa-check"></i><b>12.4.1</b> Model Checking</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="w2PGLS.html"><a href="w2PGLS.html#conclusion"><i class="fa fa-check"></i><b>12.5</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="pathanalysis.html"><a href="pathanalysis.html"><i class="fa fa-check"></i><b>13</b> Path Analysis</a>
<ul>
<li class="chapter" data-level="13.1" data-path="pathanalysis.html"><a href="pathanalysis.html#pgls-regression"><i class="fa fa-check"></i><b>13.1</b> PGLS regression</a></li>
<li class="chapter" data-level="13.2" data-path="pathanalysis.html"><a href="pathanalysis.html#variance-inflation"><i class="fa fa-check"></i><b>13.2</b> Variance Inflation</a></li>
<li class="chapter" data-level="13.3" data-path="pathanalysis.html"><a href="pathanalysis.html#path-analysis"><i class="fa fa-check"></i><b>13.3</b> Path Analysis</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="pathanalysis.html"><a href="pathanalysis.html#example"><i class="fa fa-check"></i><b>13.3.1</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="pathanalysis.html"><a href="pathanalysis.html#conclusion-1"><i class="fa fa-check"></i><b>13.4</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="convergence.html"><a href="convergence.html"><i class="fa fa-check"></i><b>14</b> Studying Convergence</a>
<ul>
<li class="chapter" data-level="14.1" data-path="convergence.html"><a href="convergence.html#indentifying-convergent-evolution"><i class="fa fa-check"></i><b>14.1</b> Indentifying convergent evolution</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="convergence.html"><a href="convergence.html#surface"><i class="fa fa-check"></i><b>14.1.1</b> SURFACE</a></li>
<li class="chapter" data-level="14.1.2" data-path="convergence.html"><a href="convergence.html#a-super-example"><i class="fa fa-check"></i><b>14.1.2</b> A Super Example</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="convergence.html"><a href="convergence.html#quantifying-convergent-evolution"><i class="fa fa-check"></i><b>14.2</b> Quantifying convergent evolution</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="convergence.html"><a href="convergence.html#the-wheatsheaf-index"><i class="fa fa-check"></i><b>14.2.1</b> The Wheatsheaf Index</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html"><i class="fa fa-check"></i><b>15</b> Tips and Tricks</a>
<ul>
<li class="chapter" data-level="15.1" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#literature-searching-with-connected-papers"><i class="fa fa-check"></i><b>15.1</b> Literature searching with <em>Connected Papers</em></a></li>
<li class="chapter" data-level="15.2" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#extracting-data-from-a-pdf-using-excel"><i class="fa fa-check"></i><b>15.2</b> Extracting data from a PDF using Excel</a></li>
<li class="chapter" data-level="15.3" data-path="tips-and-tricks.html"><a href="tips-and-tricks.html#dataeditr"><i class="fa fa-check"></i><b>15.3</b> DataEditR</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="bibliography.html"><a href="bibliography.html"><i class="fa fa-check"></i><b>16</b> Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Comparative Methods Workshops</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="bibliography" class="section level1" number="16">
<h1><span class="header-section-number">16</span> Bibliography</h1>

<div id="refs" class="references hanging-indent">
<div>
<p>Alfaro, M.E., Santini, F., Brock, C., Alamillo, H., Dornburg, A., Rabosky, D.L., Carnevale, G., and Harmon, L.J. (2009) ‘Nine Exceptional Radiations Plus High Turnover Explain Species Diversity in Jawed Vertebrates’. <em>Proceedings of the National Academy of Sciences</em> [online] 106 (32), 13410–13414. available from &lt;<a href="https://www.pnas.org/content/106/32/13410">https://www.pnas.org/content/106/32/13410</a>&gt;</p>
</div>
<div>
<p>Arbuckle, K., Bennett, C.M., and Speed, M.P. (2014) ‘A Simple Measure of the Strength of Convergent Evolution’. <em>Methods in Ecology and Evolution</em> 5 (7), 685–693</p>
</div>
<div>
<p>Arbuckle, K. and Minter, A. (2014) <em>Windex: Windex: Analysing Convergent Evolution Using the Wheatsheaf Index</em> [online] available from &lt;<a href="https://CRAN.R-project.org/package=windex">https://CRAN.R-project.org/package=windex</a>&gt;</p>
</div>
<div>
<p>Arnold, C., Matthews, L.J., and Nunn, C.L. (2010) ‘The 10kTrees Website: A New Online Resource for Primate Phylogeny’. <em>Evolutionary Anthropology: Issues, News, and Reviews</em> [online] 19 (3), 114–118. available from &lt;<a href="https://doi.org/10.1002/evan.20251">https://doi.org/10.1002/evan.20251</a>&gt;</p>
</div>
<div>
<p>Baum, D.A. and Smith, S.D. (2012) <em>Tree Thinking: An Introduction to Phylogenetic Biology</em> [online] Macmillan Learning. available from &lt;<a href="https://books.google.co.uk/books?id=zW\_ApwAACAAJ">https://books.google.co.uk/books?id=zW\_ApwAACAAJ</a>&gt;</p>
</div>
<div>
<p>Borries, C., Sandel, A.A., Koenig, A., Fernandez-Duque, E., Kamilar, J.M., Amoroso, C.R., Barton, R.A., Bray, J., Di Fiore, A., Gilby, I.C., Gordon, A.D., Mundry, R., Port, M., Powell, L.E., Pusey, A.E., Spriggs, A., and Nunn, C.L. (2016) ‘Transparency, Usability, and Reproducibility: Guiding Principles for Improving Comparative Databases Using Primates as Examples’. <em>Evolutionary Anthropology: Issues, News, and Reviews</em> 25 (5), 232–238</p>
</div>
<div>
<p>Castiglione, S., Serio, C., Mondanaro, A., Melchionna, M., Carotenuto, F., Di Febbraro, M., Profico, A., Tamagnini, D., and Raia, P. (2020) ‘Ancestral State Estimation with Phylogenetic Ridge Regression’. <em>Evolutionary Biology</em> [online] available from &lt;<a href="https://doi.org/10.1007/s11692-020-09505-x">https://doi.org/10.1007/s11692-020-09505-x</a>&gt;</p>
</div>
<div>
<p>Castiglione, S., Tesone, G., Piccolo, M., Melchionna, M., Mondanaro, A., Serio, C., Febbraro, M.D., and Raia, P. (2018) ‘A New Method for Testing Evolutionary Rate Variation and Shifts in Phenotypic Evolution.’ <em>Methods in Ecology and Evolution</em> [online] 9, 974–983. available from &lt;<a href="https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/2041-210X.12954">https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/2041-210X.12954</a>&gt;</p>
</div>
<div>
<p>Figueroa, A.D.A.G., Alex AND McKelvy (2016) ‘A Species-Level Phylogeny of Extant Snakes with Description of a New Colubrid Subfamily and Genus’. <em>PLOS ONE</em> [online] 11 (9), 1–31. available from &lt;<a href="https://doi.org/10.1371/journal.pone.0161070">https://doi.org/10.1371/journal.pone.0161070</a>&gt;</p>
</div>
<div>
<p>FitzJohn, R.G. (2012) ‘Diversitree: Comparative Phylogenetic Analyses of Diversification in R’. <em>Methods in Ecology and Evolution</em> in press</p>
</div>
<div>
<p>Fitzjohn, R.G. (2010) ‘Quantitative Traits and Diversification’. <em>Systematic Biology</em> [online] 59 (6), 619–633. available from &lt;<a href="https://liverpool.idm.oclc.org/login?url=https://search.ebscohost.com/login.aspx?direct=true&amp;db=edselc&amp;AN=edselc.2-52.0-78650666284&amp;site=eds-live&amp;scope=site">https://liverpool.idm.oclc.org/login?url=https://search.ebscohost.com/login.aspx?direct=true&amp;db=edselc&amp;AN=edselc.2-52.0-78650666284&amp;site=eds-live&amp;scope=site</a>&gt;</p>
</div>
<div>
<p>Fox, J. and Weisberg, S. (2019) <em>An R Companion to Applied Regression</em> [online] Third. Thousand Oaks CA: Sage. available from &lt;<a href="https://socialsciences.mcmaster.ca/jfox/Books/Companion/">https://socialsciences.mcmaster.ca/jfox/Books/Companion/</a>&gt;</p>
</div>
<div>
<p>Garland, J., Theodore, Dickerman, A.W., Janis, C.M., and Jones, J.A. (1993) ‘Phylogenetic Analysis of Covariance by Computer Simulation’. <em>Systematic Biology</em> [online] 42 (3), 265–292. available from &lt;<a href="https://doi.org/10.1093/sysbio/42.3.265">https://doi.org/10.1093/sysbio/42.3.265</a>&gt;</p>
</div>
<div>
<p>Gascuel, O. and Steel, M. (2014) ‘Predicting the Ancestral Character Changes in a Tree Is Typically Easier Than Predicting the Root State’. <em>Systematic Biology</em> [online] 63 (3), 421–435. available from &lt;<a href="https://doi.org/10.1093/sysbio/syu010">https://doi.org/10.1093/sysbio/syu010</a>&gt;</p>
</div>
<div>
<p>Gonzalez-Voyer, A. and Von Hardenberg, A. (2014) ‘An Introduction to Phylogenetic Path Analysis’. in <em>Modern Phylogenetic Comparative Methods and Their Application in Evolutionary Biology</em>. Springer, 201–229</p>
</div>
<div>
<p>Grafen, A. (1989) ‘The Phylogenetic Regression’. <em>Philosophical Transactions of the Royal Society of London. B, Biological Sciences</em> [online] 326 (1233), 119–157. available from &lt;<a href="https://doi.org/10.1098/rstb.1989.0106">https://doi.org/10.1098/rstb.1989.0106</a>&gt;</p>
</div>
<div>
<p>Hammill, D. (2020) <em>DataEditR: An Interactive Editor for Viewing, Entering and Editing Data</em> [online] available from &lt;<a href="https://CRAN.R-project.org/package=DataEditR">https://CRAN.R-project.org/package=DataEditR</a>&gt;</p>
</div>
<div>
<p>Harmon, L., Weir, J., Brock, C., Glor, R., and Challenger, W. (2008) ‘GEIGER: Investigating Evolutionary Radiations’. <em>Bioinformatics</em> 24, 129–131</p>
</div>
<div>
<p>Hone, D.W.E. and Benton, M.J. (2005) ‘The Evolution of Large Size: How Does Cope’s Rule Work?’ <em>Trends in Ecology &amp; Evolution</em> [online] 20 (1), 4–6. available from &lt;<a href="https://liverpool.idm.oclc.org/login?url=https://search.ebscohost.com/login.aspx?direct=true&amp;db=edselp&amp;AN=S0169534704003143&amp;site=eds-live&amp;scope=site">https://liverpool.idm.oclc.org/login?url=https://search.ebscohost.com/login.aspx?direct=true&amp;db=edselp&amp;AN=S0169534704003143&amp;site=eds-live&amp;scope=site</a>&gt;</p>
</div>
<div>
<p>Ibrahim, N., Maganuco, S., Dal Sasso, C., Fabbri, M., Auditore, M., Bindellini, G., Martill, D.M., Zouhri, S., Mattarelli, D.A., Unwin, D.M., Wiemann, J., Bonadonna, D., Amane, A., Jakubczak, J., Joger, U., Lauder, G.V., and Pierce, S.E. (2020) ‘Tail-Propelled Aquatic Locomotion in a Theropod Dinosaur’. <em>Nature</em> [online] 581 (7806), 67–70. available from &lt;<a href="https://doi.org/10.1038/s41586-020-2190-3">https://doi.org/10.1038/s41586-020-2190-3</a>&gt;</p>
</div>
<div>
<p>Ingram, T. and Mahler, D.L. (2013) ‘SURFACE: Detecting Convergent Evolution from Comparative Data by Fitting Ornstein-Uhlenbeck Models with Stepwise Akaike Information Criterion’. <em>Methods in Ecology and Evolution</em> [online] 4 (5), 416–425. available from &lt;<a href="https://doi.org/10.1111/2041-210X.12034">https://doi.org/10.1111/2041-210X.12034</a>&gt;</p>
</div>
<div>
<p>Jones, K.E., Bielby, J., Cardillo, M., Fritz, S.A., O’Dell, J., Orme, D.L., Safi, K., Sechrest, W., Boakes, E.H., Carbone, C., Connolly, C., Cutts, M.J., Foster, J.K., Grenyer, R., Habib, M., Plaster, C.A., Price, S.A., Rigby, E.A., Rist, J., Teacher, A., Bininda-Emonds, O.R.P., Gittleman, J.L., Mace, G.M., and Purvis, A. (2009) ‘PanTHERIA: A Species-Level Database of Life History, Ecology, and Geography of Extant and Recently Extinct Mammals’. <em>Ecology</em> 90 (9), 2648–2648</p>
</div>
<div>
<p>Joy, J.B., Liang, R.H., McCloskey, R.M., Nguyen, T., and Poon, A.F.Y. (2016) ‘Ancestral Reconstruction’. <em>PLOS Computational Biology</em> [online] 12 (7), e1004763. available from &lt;<a href="https://doi.org/10.1371/journal.pcbi.1004763">https://doi.org/10.1371/journal.pcbi.1004763</a>&gt;</p>
</div>
<div>
<p>Kingsolver, J.G. and Pfennig, D.W. (2004) ‘Individual-Level Selection as a Cause of Cope’s Rule of Phyletic Size Increase’. <em>Evolution</em> [online] 58 (7), 1608. available from &lt;<a href="https://liverpool.idm.oclc.org/login?url=https://search.ebscohost.com/login.aspx?direct=true&amp;db=edsjsr&amp;AN=edsjsr.3449385&amp;site=eds-live&amp;scope=site">https://liverpool.idm.oclc.org/login?url=https://search.ebscohost.com/login.aspx?direct=true&amp;db=edsjsr&amp;AN=edsjsr.3449385&amp;site=eds-live&amp;scope=site</a>&gt;</p>
</div>
<div>
<p>Louca, S. and Pennell, M.W. (2020) ‘Extant Timetrees Are Consistent with a Myriad of Diversification Histories’. <em>Nature</em> [online] 580 (7804), 502–505. available from &lt;<a href="https://doi.org/10.1038/s41586-020-2176-1">https://doi.org/10.1038/s41586-020-2176-1</a>&gt;</p>
</div>
<div>
<p>McGowen, M.R., Montgomery, S.H., Clark, C., and Gatesy, J. (2011) ‘Phylogeny and Adaptive Evolution of the Brain-Development Gene Microcephalin (Mcph1) in Cetaceans’. <em>BMC Evolutionary Biology</em> [online] 11 (1), 98. available from &lt;<a href="https://doi.org/10.1186/1471-2148-11-98">https://doi.org/10.1186/1471-2148-11-98</a>&gt;</p>
</div>
<div>
<p>Montgomery, S.H., Capellini, I., Barton, R.A., and Mundy, N.I. (2010) ‘Reconstructing the Ups and Downs of Primate Brain Evolution: Implications for Adaptive Hypotheses and Homo Floresiensis’. <em>BMC Biology</em> [online] 8 (1), 9. available from &lt;<a href="https://doi.org/10.1186/1741-7007-8-9">https://doi.org/10.1186/1741-7007-8-9</a>&gt;</p>
</div>
<div>
<p>Montgomery, S.H., Geisler, J.H., McGowen, M.R., Fox, C., Marino, L., and Gatesy, J. (2013) <em>The Evolutionary History of Cetacean Brain and Body Size</em>. [online] 67 (11), 3339–3353. available from &lt;<a href="www.jstor.org/stable/24032748">www.jstor.org/stable/24032748</a>&gt;</p>
</div>
<div>
<p>Morgan, M. (2019) <em>BiocManager: Access the Bioconductor Project Package Repository</em> [online] available from &lt;<a href="https://CRAN.R-project.org/package=BiocManager">https://CRAN.R-project.org/package=BiocManager</a>&gt;</p>
</div>
<div>
<p>Morlon, H., Lewitus, E., Condamine, F., Manceau, M., Clavel, J., and Drury, J. (2016) ‘RPANDA: An R Package for Macroevolutionary Analyses on Phylogenetic Trees’. <em>Methods in Ecology and Evolution</em> [online] 7, 589–597. available from &lt;<a href="https://CRAN.R-project.org/package=RPANDA">https://CRAN.R-project.org/package=RPANDA</a>&gt;</p>
</div>
<div>
<p>Neal, R.M. (2003) ‘Slice Sampling’. <em>Annals of Statistics</em> [online] 31 (3), 705–741. available from &lt;<a href="https://liverpool.idm.oclc.org/login?url=https://search.ebscohost.com/login.aspx?direct=true&amp;db=edselc&amp;AN=edselc.2-52.0-1642370803&amp;site=eds-live&amp;scope=site">https://liverpool.idm.oclc.org/login?url=https://search.ebscohost.com/login.aspx?direct=true&amp;db=edselc&amp;AN=edselc.2-52.0-1642370803&amp;site=eds-live&amp;scope=site</a>&gt;</p>
</div>
<div>
<p>Nunn, C.L. (2011) <em>The Comparative Approach in Evolutionary Anthropology and Biology</em> [online] University of Chicago Press. available from &lt;<a href="https://books.google.co.uk/books?id=qj4cSzJGQJAC">https://books.google.co.uk/books?id=qj4cSzJGQJAC</a>&gt;</p>
</div>
<div>
<p>Nyakatura, K. and Bininda-Emonds, O.R. (2012) ‘Updating the Evolutionary History of Carnivora (Mammalia): A New Species-Level Supertree Complete with Divergence Time Estimates’. <em>BMC Biology</em> [online] 10 (1), 12. available from &lt;<a href="https://doi.org/10.1186/1741-7007-10-12">https://doi.org/10.1186/1741-7007-10-12</a>&gt;</p>
</div>
<div>
<p>Orme, D., Freckleton, R., Thomas, G., Petzoldt, T., Fritz, S., Isaac, N., and Pearse, W. (2018) <em>Caper: Comparative Analyses of Phylogenetics and Evolution in R</em> [online] available from &lt;<a href="https://CRAN.R-project.org/package=caper">https://CRAN.R-project.org/package=caper</a>&gt;</p>
</div>
<div>
<p>Paradis, E. and Schliep, K. (2018) ‘Ape 5.0: An Environment for Modern Phylogenetics and Evolutionary Analyses in R’. <em>Bioinformatics</em> 35, 526–528</p>
</div>
<div>
<p>Parker, H.G., Dreger, D.L., Rimbault, M., Davis, B.W., Mullen, A.B., Carpintero-Ramirez, G., and Ostrander, E.A. (2017) ‘Genomic Analyses Reveal the Influence of Geographic Origin, Migration, and Hybridization on Modern Dog Breed Development’. <em>Cell Reports</em> [online] 19 (4), 697–708. available from &lt;<a href="http://www.sciencedirect.com/science/article/pii/S2211124717304564">http://www.sciencedirect.com/science/article/pii/S2211124717304564</a>&gt;</p>
</div>
<div>
<p>Pedersen, T.L. (2019) <em>Ggforce: Accelerating ’Ggplot2’</em> [online] available from &lt;<a href="https://CRAN.R-project.org/package=ggforce">https://CRAN.R-project.org/package=ggforce</a>&gt;</p>
</div>
<div>
<p>Peters, R.S., Krogmann, L., Mayer, C., Donath, A., Gunkel, S., Meusemann, K., Kozlov, A., Podsiadlowski, L., Petersen, M., Lanfear, R., Diez, P.A., Heraty, J., Kjer, K.M., Klopfstein, S., Meier, R., Polidori, C., Schmitt, T., Liu, S., Zhou, X., Wappler, T., Rust, J., Misof, B., and Niehuis, O. (2017) ‘Evolutionary History of the Hymenoptera’. <em>Current Biology</em> [online] 27 (7), 1013–1018. available from &lt;<a href="https://doi.org/10.1016/j.cub.2017.01.027">https://doi.org/10.1016/j.cub.2017.01.027</a>&gt;</p>
</div>
<div>
<p>Rabosky, D., Grundler, M., Anderson, C., Title, P., Shi, J., Brown, J., Huang, H., and Larson, J. (2014) ‘BAMMtools: An R Package for the Analysis of Evolutionary Dynamics on Phylogenetic Trees’. <em>Methods in Ecology and Evolution</em> 5, 701–707</p>
</div>
<div>
<p>Rabosky, D.L. (2014) ‘Automatic Detection of Key Innovations, Rate Shifts, and Diversity-Dependence on Phylogenetic Trees’. <em>PLOS ONE</em> [online] 9 (2), 1–15. available from &lt;<a href="https://doi.org/10.1371/journal.pone.0089543">https://doi.org/10.1371/journal.pone.0089543</a>&gt;</p>
</div>
<div>
<p>R Core Team (2019) <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing</p>
</div>
<div>
<p>Reader, S.M., Hager, Y., and Laland, K.N. (2011) ‘The Evolution of Primate General and Cultural Intelligence’. <em>Philosophical Transactions of the Royal Society B: Biological Sciences</em> [online] 366 (1567), 1017–1027. available from &lt;<a href="https://royalsocietypublishing.org/doi/abs/10.1098/rstb.2010.0342">https://royalsocietypublishing.org/doi/abs/10.1098/rstb.2010.0342</a>&gt;</p>
</div>
<div>
<p>Revell, L.J. (2012) ‘Phytools: An R Package for Phylogenetic Comparative Biology (and Other Things).’ <em>Methods in Ecology and Evolution</em> 3, 217–223</p>
</div>
<div>
<p>R Hackathon et al. (2019) <em>Phylobase: Base Package for Phylogenetic Structures and Comparative Data</em> [online] available from &lt;<a href="https://CRAN.R-project.org/package=phylobase">https://CRAN.R-project.org/package=phylobase</a>&gt;</p>
</div>
<div>
<p>Serio, C., Castiglione, S., Tesone, G., Piccolo, M., Melchionna, M., Mondanaro, A., Febbraro, M.D., and Raia, P. (2019) ‘Macroevolution of Toothed Whales Exceptional Relative Brain Size.’ <em>Evolutionary Biology</em> [online] 46, 332–342. available from &lt;<a href="https://link.springer.com/article/10.1007/s11692-019-09485-7">https://link.springer.com/article/10.1007/s11692-019-09485-7</a>&gt;</p>
</div>
<div>
<p>Shipley, B. (2016) <em>Cause and Correlation in Biology: A User’s Guide to Path Analysis, Structural Equations and Causal Inference with R</em>. Cambridge University Press</p>
</div>
<div>
<p>Shipley, B. (2000) ‘A New Inferential Test for Path Models Based on Directed Acyclic Graphs’. <em>Structural Equation Modeling</em> 7 (2), 206–218</p>
</div>
<div>
<p>Slater, G.J., Harmon, L.J., and Alfaro, M.E. (2012) ‘Integrating Fossils with Molecular Phylogenies Improves Inference of Trait Evolution’. <em>Evolution</em> [online] 66 (12), 3931–3944. available from &lt;<a href="https://doi.org/10.1111/j.1558-5646.2012.01723.x">https://doi.org/10.1111/j.1558-5646.2012.01723.x</a>&gt;</p>
</div>
<div>
<p>Stadler, T. (2019) <em>TreeSim: Simulating Phylogenetic Trees</em> [online] available from &lt;<a href="https://CRAN.R-project.org/package=TreeSim">https://CRAN.R-project.org/package=TreeSim</a>&gt;</p>
</div>
<div>
<p>Stankowich, T., Caro, T., and Cox, M. (2011) ‘Bold Coloration and the Evolution of Aposematism in Terrestrial Carnivores’. <em>Evolution</em> [online] 65 (11), 3090. available from &lt;<a href="https://liverpool.idm.oclc.org/login?url=https://search.ebscohost.com/login.aspx?direct=true&amp;db=edsjsr&amp;AN=edsjsr.41317030&amp;site=eds-live&amp;scope=site">https://liverpool.idm.oclc.org/login?url=https://search.ebscohost.com/login.aspx?direct=true&amp;db=edsjsr&amp;AN=edsjsr.41317030&amp;site=eds-live&amp;scope=site</a>&gt;</p>
</div>
<div>
<p>Stumpke, H. (1957) <em>The Snouters: Form and Life of the Rhinogrades</em>. Stuttgart: Gustav Fischer Verlag</p>
</div>
<div>
<p>Thewissen, J.G.M., Cooper, L.N., George, J.C., and Bajpai, S. (2009) ‘From Land to Water: The Origin of Whales, Dolphins, and Porpoises’. <em>Evolution: Education and Outreach</em> [online] 2 (2), 272–288. available from &lt;<a href="https://doi.org/10.1007/s12052-009-0135-2">https://doi.org/10.1007/s12052-009-0135-2</a>&gt;</p>
</div>
<div>
<p>Urbanek, S. (2019) <em>Jpeg: Read and Write Jpeg Images</em> [online] available from &lt;<a href="https://CRAN.R-project.org/package=jpeg">https://CRAN.R-project.org/package=jpeg</a>&gt;</p>
</div>
<div>
<p>van der Bijl, W. (2018) ‘Phylopath: Easy Phylogenetic Path Analysis in R.’ <em>PeerJ</em> 6, e4718</p>
</div>
<div>
<p>von Hardenberg, A. and Gonzalez-Voyer, A. (2013) ‘Disentangling Evolutionary Cause-Effect Relationships with Phylogenetic Confirmatory Path Analysis’. <em>Evolution</em> 67 - 2, 378–387</p>
</div>
<div>
<p>Webster, A.J. and Purvis, A. (2002) ‘Testing the Accuracy of Methods for Reconstructing Ancestral States of Continuous Characters’. <em>Proceedings of the Royal Society of London. Series B: Biological Sciences</em> [online] 269 (1487), 143–149. available from &lt;<a href="https://doi.org/10.1098/rspb.2001.1873">https://doi.org/10.1098/rspb.2001.1873</a>&gt;</p>
</div>
<div>
<p>Wickham, H. (2017) <em>Tidyverse: Easily Install and Load the ’Tidyverse’</em> [online] available from &lt;<a href="https://CRAN.R-project.org/package=tidyverse">https://CRAN.R-project.org/package=tidyverse</a>&gt;</p>
</div>
<div>
<p>Wickham, H. (2016) <em>Ggplot2: Elegant Graphics for Data Analysis</em> [online] Springer-Verlag New York. available from &lt;<a href="https://ggplot2.tidyverse.org">https://ggplot2.tidyverse.org</a>&gt;</p>
</div>
<div>
<p>Yu, G. (2019) <em>Ggimage: Use Image in ’Ggplot2’</em> [online] available from &lt;<a href="https://CRAN.R-project.org/package=ggimage">https://CRAN.R-project.org/package=ggimage</a>&gt;</p>
</div>
<div>
<p>Yu, G., Lam, T.T.-Y., Zhu, H., and Guan, Y. (2018) ‘Two Methods for Mapping and Visualizing Associated Data on Phylogeny Using Ggtree.’ <em>Molecular Biology and Evolution</em> [online] 35 (2), 3041–3043. available from &lt;<a href="https://doi.org/10.1093/molbev/msy194">https://doi.org/10.1093/molbev/msy194</a>&gt;</p>
</div>
<div>
<p>Yu, G., Smith, D., Zhu, H., Guan, Y., and Lam, T.T.-Y. (2017) ‘Ggtree: An R Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data.’ <em>Methods in Ecology and Evolution</em> [online] 8 (1), 28–36. available from &lt;<a href="http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract">http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12628/abstract</a>&gt;</p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="tips-and-tricks.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
